clients/src/main/java/org/apache/kafka/common/utils/Shell.java

```diff
@@ -0,0 +1,335 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package org.apache.kafka.common.utils;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A base class for running a Unix command.
+ *
+ * <code>Shell</code> can be used to run unix commands like <code>du</code> or
+ * <code>df</code>.
+ */
+abstract public class Shell {
+
+    private static final Logger LOG = LoggerFactory.getLogger(Shell.class);
+
+    private int exitCode;
+
+    /** return an array containing the command name & its parameters */
+    protected abstract String[] execString();
+
+    /** Parse the execution result */
+    protected abstract void parseExecResult(BufferedReader lines)
+        throws IOException;
+
+    /**Time after which the executing script would be timedout*/
+    protected long timeOutInterval = 0L;
+
+    private long    interval;   // refresh interval in msec
+    private long    lastTime;   // last time the command was performed
+    private Process process; // sub process used to execute the command
+
+    /**If or not script finished executing*/
+    private volatile AtomicBoolean completed;
+
+    public static final Time TIME = new SystemTime();
+
+    public Shell() {
+        this(0L);
+    }
+
+    /**
+     * @param interval the minimum duration to wait before re-executing the
+     *        command.
+     */
+    public Shell(long interval) {
+        this.interval = interval;
+        this.lastTime = (interval < 0) ? 0 : -interval;
```

Comment:
As far as I can see, we don't need to support interval. I have removed it locally as it simplifies the class.